# aDNA Data Analysis II: assessment and evolutionary analysis of ancient Mycobacterium leprae genomes

## Scenario

You have received NGS *Mycobacterium leprae* capture data for three skeletons from Sant Llàtzer, a medieval leprosarium in Barcelona: **UF104**, **UF703**, **UF801**. 

The genetic diversity of *M. leprae* strains across medieval Europe and within leprosaria was high. No one has recovered ancient *M. leprae* genomes from Spain, so you would like to know if leprosy infections at Sant Llátzer were seeded by a diversity of lineages.  

## Part 1: Using Qualimap to determin reference coverage and depth

Since you have already assessed these raw data (*aDNA Data Analysis I*), you will begin this practical by using **Qualimap** to assess genome coverage and depth from `.bam` files that were generated by mapping reads to the *Mycobacterium leprae* reference genome. 

I have run the tool for you, and you have the outputs: a `.pdf` report and a `.txt` file for each sample.

Open [Sant_Llàtzer_M.leprae_mapping_report](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/M.leprae%20mapping%20report.tsv) in a new tab (right click). Since you cannot edit this file directly, download it using the download icon in the top right corner. The extension is `.tsv`, which stands for "tab-separated values." Open this file using excel to be able to edit the spreadsheet.

[**Qualimap**](http://qualimap.conesalab.org/) is a tool that assesses genome coverage and depth from `.bam` files. I have run the tool for you and generated outputs.

|  Qualimap `.pdf` report | Qualimap `.txt` report |
|-------------|-------------|
| [UF104.pdf](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF104_report.pdf) | [UF104.txt](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF104_genome_results.txt) |
|  [UF703.pdf](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF703_report.pdf)  | [UF703.txt](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF703_genome_results.txt) |
| [UF801.pdf](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF801_report.pdf)  | [UF801.txt](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF801_genome_results.txt) |

Open the `.pdf` reports for your samples.

Fill in the following columns in the mapping report:

- `Q25 mapped reads after duplicate removal` 
- `Avg length of mapped Q25 reads`
- `Mean Cov`
- `SD Cov`  

Now that you have read through the `.pdf` reports, open the **Qualimap** `.txt` files. Much — but not all — of the information from the `.pdf` is reported here as well. 

Text file outputs are useful because you can write code that will loop through `.txt` files and pull out the information you need. Most geneticists automate their mapping reports so that they do not have to manually extract each piece of information.  

Complete the next two columns in your **Sant Llátzer mapping report** using the **Qualimap** `.txt` files.:

- `% ref covered at >=1x`  
- `% ref covered at >=5x`  

Some of the metrics you need to assess your data are not generated by a tool, so you need to generate them yourself. **Cluster factor**, also known as **library complexity**, is informative about the number of unique fragments in your library. It tells you on average how many times a read was sequenced. **Library complexity** refers to how many unique DNA fragments are in your sequencing data. One way to measure this is by looking at the number or percentage of **duplicate reads**, which are reads that start and end at the same positions when aligned to a reference genome. They occur during the multiple rounds of PCR necessary to prepare a library for sequencing.

For example:
- **Cluster values over 2** indicate that, on average, every fragment has been sequenced twice — meaning the library is likely exhausted of DNA from this organism.
- Depending on how deeply you sequence and whether you’ve done targeted enrichment, **cluster factors can be huge**, ranging anywhere between **20 and 30**.

In your mapping report, **Calculate cluster factor** for each sample using the formula in the column header and the values in the columns you’ve already filled out.

**Endogenous DNA Frequency**

Endogenous DNA frequency can be used alongside cluster factor to decide whether to sequence your library more deeply or to assess how efficient your capture enrichment was. Since these *M. leprae* data were obtained through targeted capture enrichment, endogenous DNA frequency will show how effective the capture experiment was.

**Calculate the frequency of endogenous DNA** (*M. leprae* in this case) for each sample using the formula in the column header cell and the values in the columns you’ve already filled out.

## Part 2: MapDamage plots

Review: 

Remember that **deamination** results in the chemical modification (loss of an amine group) of cytosine such that it becomes uracil (Fig. 1). This is probematic because uracils are complementary to adenine, not guanine like the original cytosine base. Therefore, in subsequent PCRs, the original cytosine becomes a thymine in the forward direction. In the reverse direction of the fragment, the guanine becomes an adenine base.

<img src="https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/images/depurination.png" alt="deamination" width="1000">

*Figure 1: Depiction of depurination and deamination (from [Orlando et al., 2021](https://www.nature.com/articles/s43586-020-00011-0)).*

Uracils can be removed by treating the DNA extracts with USER enzyme as depicted below (Fig. 2). USER enzyme is a combination of two enzymes, principally uracil DNA glycosylase (UDG). 

<img src="https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/images/USER_enzyme.png" alt="USER" width="700">

*Figure 2: Depiction of USER enzyme treatment (from [NEB](https://www.neb.com/en-gb/products/m5505-user-enzyme)).*

You may remember, however, that deamination can be useful for authenticating DNA as ancient, so we may not want to remove all uracils. [Rohland and colleauges (2015)](https://royalsocietypublishing.org/doi/10.1098/rstb.2013.0624) developed a method termed "partial UDG treatment" that excises uracils except those in the terminal positions of the fragments (Fig. 3). This is useful becuase it perserves the damage signature but isolates it to the terminal ~2 bases, whereas normally it is distributed deeper into the fragment. 

<img src="https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/images/effect-of-UDG-treatment.png" alt="partialUDG" width="700">

*Figure 3: Depiction of full versus partial USER (UDG) enzyme treatment. **A** Full UDG treatment depiction and **B** resulting damange plot. **C** Partial UDG treatment and **D** resulting damage plot (from [Introduction to ancient metagenomics resource Fig 2.20](https://www.spaam-community.org/intro-to-ancient-metagenomics-book/introduction-to-ancient-dna.html#how-dna-degrades)).*


There are several tools for generating aDNA "damage plots," but here we are using [**MapDamage**](https://ginolhac.github.io/mapDamage/) because it also **rescales base qualities**, which other damage profiling tools do not. 

**Open the MapDamage outputs**

|  MapDamage `.pdf` reports |
|-------|
| [UF104-MapDamage.pdf](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/2.MapDamage%20outputs/UF104_Fragmisincorporation_plot.pdf) |
| [UF703-MapDamage.pdf](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/2.MapDamage%20outputs/UF703_Fragmisincorporation_plot.pdf) |
| [UF801-MapDamage.pdf](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/2.MapDamage%20outputs/UF801_Fragmisincorporation_plot.pdf) |

Look at the **fragment misincorporation plots** for each sample:

   - The **four plots at the top** show base frequencies (A, C, T, and G) within the read (inside the grey square) and outside the read.
   - These plots illustrate patterns of **depurination** — a type of aDNA damage where purines (A and G) are lost over time (Fig. 1).

The **two plots at the bottom** are the most recognizable damage plots cause by **deamination** (Fig. 1):
   
   - The **red line** charts **C to T substitutions**.
   - The **blue line** charts **G to A substitutions**.


### Tasks:

- **Estimate the values** for the last four columns in your ***M. leprae* mapping report** using the bottom two plots.
- **Assess authenticity**: Based on these plots, do all the samples show patterns consistent with **authentic ancient DNA**?
- What are the multiple ways you could interpret no sign of deamination damage in the MapDamage plot?

