# aDNA Data Analysis II: assessment and evolutionary analysis of ancient Mycobacterium leprae genomes

## Scenario

You have received NGS *M. leprae* capture data for three skeletons from Sant Llàtzer, a medieval leprosarium in Barcelona: **UF104**, **UF703**, **UF801**. 

The genetic diversity of *M. leprae* strains across medieval Europe and within leprosaria was high. No one has recovered ancient *M. leprae* genomes from Spain, so you would like to know if leprosy infections at Sant Llátzer were seeded by a diversity of lineages.  

## Part 1: Using Qualimap to determin reference coverage and depth

Since you have already assessed these raw data (*aDNA Data Analysis I*), you will begin this practical by using **Qualimap** to assess genome coverage and depth from `.bam` files that were generated by mapping reads to the *Mycobacterium leprae* reference genome. 

I have run the tool for you, and you have the outputs: a `.pdf` report and a `.txt` file for each sample.

Open [Sant_Llàtzer_M.leprae_mapping_report](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/M.leprae%20mapping%20report.tsv) in a new tab (right click). Since you cannot edit this file directly, download it using the download icon in the top right corner. The extension is `.tsv`, which stands for "tab-separated values." Open this file using excel to be able to edit the spreadsheet.

[**Qualimap**](http://qualimap.conesalab.org/) is a tool that assesses genome coverage and depth from `.bam` files. I have run the tool for you and generated outputs.

| Sample |  Qualimap `.pdf` report | Qualimap `.txt` report |
|--------|-------------|-------------|
| UF104 | [UF104.pdf](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF104_report.pdf) | [UF104.txt](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF104_genome_results.txt) |
| UF703 | [UF703.pdf](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF703_report.pdf)  | [UF703.txt](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF703_genome_results.txt) |
| UF801 | [UF801.pdf](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF801_report.pdf)  | [UF801.txt](https://github.com/Kelzor/Introductory-ancient-DNA-practical/blob/main/aDNA%20Data%20Analysis%20II%20-%20M.%20leprae/1.Qualimap%20outputs/UF801_genome_results.txt) |

Open the `.pdf` and `.txt` reports for your samples.  

Fill in the following columns in the mapping report:

`Q25 mapped reads after duplicate removal` 

`Avg length of mapped Q25 reads`  

`Mean Cov` 

`SD Cov`  


